src/mainwindow.cpp

```diff
@@ -740,3 +738,16 @@ void MainWindow::addNewTab()
     else
         consoleTabulator->addNewTab();
 }
+
+void MainWindow::onCurrentTitleChanged(int index)
+{
+    QString title;
+    QIcon icon;
+    if (-1 != index)
+    {
+        title = consoleTabulator->tabText(index);
+        icon = consoleTabulator->tabIcon(index);
+    }
+    setWindowTitle(title.isEmpty() ? QStringLiteral("QTerminal") : title);
+    setWindowIcon(icon.isNull() ? QIcon::fromTheme("utilities-terminal") : icon);
```

Comment:
So you propose, not to change the qterminal's main icon based on icon of selected terminal? Sounds reasonable... but IMO reasonable is it the other way round. Maybe we should make it configurable. What do you mean?